memphis/memphis.py

```diff
@@ -0,0 +1,412 @@
+            max_msg_deliveries (int, optional): max number of message deliveries, by default is 10.
+
+        Returns:
+            object: consumer
+        """
+        try:
+            if not self.is_connection_active:
+                raise Exception("Connection is dead")
+            cg = consumer_name if not consumer_group else consumer_group
+            http_request("POST", 'http://'+self.host+':' + str(self.management_port)+'/api/consumers/createConsumer', headers={"Authorization": "Bearer " + self.access_token}, body_params={
+                "name": consumer_name, "station_name": station_name, "connection_id": self.connection_id, "consumer_type": "application", "consumers_group": cg, "max_ack_time_ms": max_ack_time_ms, "max_msg_deliveries": max_msg_deliveries})
+            return Consumer(self, station_name, consumer_name, cg, pull_interval_ms, batch_size, batch_max_time_to_wait_ms, max_ack_time_ms, max_msg_deliveries)
+        except Exception as e:
+            raise Exception(e)
+
+
+class Factory:
+    def __init__(self, connection, name):
+        self.connection = connection
+        self.name = name.lower()
+
+    def destroy(self):
+        """Destroy the factory.
+        """
+        try:
+            http_request("DELETE", 'http://'+self.connection.host+':'+str(self.connection.management_port)+'/api/factories/removeFactory', headers={
+                "Authorization": "Bearer " + self.connection.access_token}, body_params={"factory_name": self.name})
+        except Exception as e:
+            raise Exception(e)
+
+
+class Station:
+    def __init__(self, connection, name):
+        self.connection = connection
+        self.name = name.lower()
+
+    def destroy(self):
+        """Destroy the station.
+        """
+        try:
+            http_request("DELETE", 'http://'+self.connection.host+':'+str(self.connection.management_port)+'/api/stations/removeStation', headers={
+                "Authorization": "Bearer " + self.connection.access_token}, body_params={"station_name": self.name})
+        except Exception as e:
+            raise Exception(e)
+
+
+class Producer:
+    def __init__(self, connection, producer_name, station_name):
+        self.connection = connection
+        self.producer_name = producer_name
+        self.station_name = station_name
+
+    async def produce(self, message, ack_wait_sec=15):
+        """Produces a message into a station.
+
+        Args:
+            message (Uint8Array): message to send into the station.
+            ack_wait_sec (int, optional): max time in seconds to wait for an ack from memphis. Defaults to 15.
+
+        Raises:
+            Exception: _description_
+            Exception: _description_
+        """
+        try:
+            await self.connection.broker_connection.publish(self.station_name + ".final", message, headers={
+                "Nats-Msg-Id": str(uuid.uuid4()), "ackWait": str(ack_wait_sec), "producedBy": self.producer_name})
```

Comment:
I wonder if it works sending ackWait header

Reply:
Fixed